src/ripple/nodestore/backend/NuDBFactory.cpp

```diff
@@ -174,10 +183,20 @@ class NuDBBackend : public Backend
             nudb::error_code ec;
             db_.close(ec);
             if (ec)
+            {
+                // Log to make sure the nature of the error gets to the user.
+                JLOG(j_.fatal()) << "NuBD close() failed: " << ec.message();
                 Throw<nudb::system_error>(ec);
```

Comment:
Since the error is getting logged, and the nearest catch block doesn't do anything with this error, what's the reason for still throwing? If we get rid of it seems like we could also eliminate the `try/catch` block in the destructor. This would also match the approach being used for the following call to `remove_all`.

Reply:
The reason I left the `throw` was so behavior is unchanged for any calls to `close()` other than the one from the destructor.  I did not audit all calls to `close()`, but I assumed there were some because `close()` is a public method of `Backend`.  Note the comment in Backend.cpp:
```
    /** Close the backend.
        This allows the caller to catch exceptions.
    */
    virtual void
    close() = 0;
```

Reply:
We would need to prove that no other client of `close()` is depending on the exception.  This isn't impossible, but is a little tricky considering this is a public virtual.

Reply:
Preserving the existing throw sounds good